### PR TITLE
refactor key updates on host basis vice node

### PR DESCRIPTION
**Pull request by mattkasun**
## Describe your changes
new endpoints for host key updates
deletion of network key update endpoint

## Provide testing steps
see GRA-1529

## Checklist before requesting a review
- [x] My changes affect only 10 files or less.
- [ ] I have performed a self-review of my code and tested it.
- [ ] If it is a new feature, I have added thorough tests, my code is <= 1450 lines.
- [x] If it is a bugfix, my code is <= 200 lines.
- [x] My functions are <= 80 lines.
- [ ] I have had my code reviewed by a peer.
- [ ] My unit tests pass locally.
- [ ] Netmaker is awesome.

### CODE REVIEW

**Review by Aceix — APPROVED** [review]

**Review by theguy951357 — APPROVED** [review]
pr passes
critical list passes

**Review by theguy951357 — APPROVED** [review]
pr steps pass
critical list passes

File: mq/handlers.go [hunk 1]
```diff
@@ -117,6 +118,27 @@ func UpdateHost(client mqtt.Client, msg mqtt.Message) {
...
+					},
+				},
+			}
+			data, err := json.Marshal(&peerUpdate)
+			if err != nil {
+				logger.Log(2, "json error", err.Error())
+			}
+			hosts := logic.GetRelatedHosts(hostUpdate.Host.ID.String())
+			for _, host := range hosts {
+				publish(&host, fmt.Sprintf("peers/host/%s/%s", host.ID.String(), servercfg.GetServer()), data)
```
Comment (0xdcarns) [inline mq/handlers.go:138]: servercfg.GetServer() performs a small amount of logic, should be set to variable outside of loop. `serverName := servercfg.GetServer()`

**Review by 0xdcarns — CHANGES_REQUESTED** [review]

File: mq/handlers.go (hunk 1 above)
Comment (0xdcarns) [reply mq/handlers.go:138]: also resolve conflicts

**Review by 0xdcarns — APPROVED** [review]

File: controllers/network.go [hunk 2]
```diff
@@ -70,7 +69,6 @@ func getNetworks(w http.ResponseWriter, r *http.Request) {
 	}
 
 	logger.Log(2, r.Header.Get("user"), "fetched networks.")
-	logic.SortNetworks(allnetworks[:])
```
Comment (0xdcarns) [inline controllers/network.go:73]: put this back

File: logic/networks.go [hunk 3]
```diff
@@ -588,45 +578,3 @@ func NetworkExists(name string) (bool, error) {
...
-				return err
-			}
-			database.Insert(node.ID.String(), string(data), database.NODES_TABLE_NAME)
-		}
-	}
-	return nil
-}
-
-// SortNetworks - Sorts slice of Networks by their NetID alphabetically with numbers first
-func SortNetworks(unsortedNetworks []models.Network) {
```
Comment (0xdcarns) [inline logic/networks.go:628]: put this function back

**Review by 0xdcarns — CHANGES_REQUESTED** [review]

**Review by 0xdcarns — APPROVED** [review]

**Pull request merged**